azurerm/resource_arm_dev_test_lab_schedule.go

```diff
@@ -0,0 +1,447 @@
+	if err != nil {
+		return err
+	}
+
+	read, err := client.Get(ctx, resGroup, devTestLabName, name, "")
+	if err != nil {
+		return err
+	}
+
+	if read.ID == nil {
+		return fmt.Errorf("Cannot read  Dev Test Lab Schedule %s (resource group %s) ID", name, resGroup)
+	}
+
+	d.SetId(*read.ID)
+
+	return resourceArmDevTestLabSchedulesRead(d, meta)
+}
+
+func resourceArmDevTestLabSchedulesRead(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*ArmClient).devTestLabs.LabSchedulesClient
+	ctx := meta.(*ArmClient).StopContext
+
+	id, err := parseAzureResourceID(d.Id())
+	if err != nil {
+		return err
+	}
+	resGroup := id.ResourceGroup
+	devTestLabName := id.Path["labs"]
+	name := id.Path["schedules"]
+
+	resp, err := client.Get(ctx, resGroup, devTestLabName, name, "")
+
+	if err != nil {
+		if utils.ResponseWasNotFound(resp.Response) {
+			d.SetId("")
+			return nil
+		}
+		return fmt.Errorf("Error making Read request on Dev Test Lab Schedule %s: %s", name, err)
+	}
+
+	d.Set("name", resp.Name)
+	if location := resp.Location; location != nil {
+		d.Set("location", azure.NormalizeLocation(*location))
+	}
+	d.Set("lab_name", devTestLabName)
+	d.Set("resource_group_name", resGroup)
+
+	if props := resp.ScheduleProperties; props != nil {
+
+		d.Set("time_zone_id", props.TimeZoneID)
+
+		d.Set("status", string(props.Status))
+
+		if err := d.Set("weekly_recurrence", flattenAzureRmDevTestLabScheduleRecurrenceWeekly(props.WeeklyRecurrence)); err != nil {
+			return fmt.Errorf("Error setting `weeklyRecurrence`: %#v", err)
+		}
+
+		if err := d.Set("daily_recurrence", flattenAzureRmDevTestLabScheduleRecurrenceDaily(props.DailyRecurrence)); err != nil {
+			return fmt.Errorf("Error setting `dailyRecurrence`: %#v", err)
+		}
+
+		if err := d.Set("hourly_recurrence", flattenAzureRmDevTestLabScheduleRecurrenceHourly(props.HourlyRecurrence)); err != nil {
+			return fmt.Errorf("Error setting `dailyRecurrence`: %#v", err)
+		}
+
+		if err := d.Set("notification_settings", flattenAzureRmDevTestLabScheduleNotificationSettings(props.NotificationSettings)); err != nil {
+			return fmt.Errorf("Error setting `notificationSettings`: %#v", err)
+		}
+	}
+
+	flattenAndSetTags(d, resp.Tags)
+
+	return nil
+}
+
+func resourceArmDevTestLabSchedulesDelete(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*ArmClient).vmExtensionClient
+	ctx := meta.(*ArmClient).StopContext
+
+	id, err := parseAzureResourceID(d.Id())
+	if err != nil {
+		return err
+	}
+	resGroup := id.ResourceGroup
+	name := id.Path["schedules"]
+	devTestLabName := id.Path["labs"]
+
+	future, err := client.Delete(ctx, resGroup, devTestLabName, name)
+	if err != nil {
+		return err
+	}
+
+	return future.WaitForCompletionRef(ctx, client.Client)
+}
+
+func expandArmDevTestLabScheduleRecurrenceDaily(d *schema.ResourceData) *dtl.DayDetails {
+	dailyRecurrenceConfigs := d.Get("daily_recurrence").([]interface{})
```

Comment:
We should just pass in the value and not do `d.Get` here

Reply:
Done